File: app/android/app/src/main/java/com/gonezo/multiplatform/audioextraction/application/ResultAssembler.java
package com.gonezo.multiplatform.audioextraction.application;

import com.gonezo.multiplatform.audioextraction.contract.ExtractionResult;
import com.gonezo.multiplatform.audioextraction.domain.model.ExecutionPlan;
import com.gonezo.multiplatform.audioextraction.domain.model.ResolvedField;
import com.gonezo.multiplatform.audioextraction.domain.model.Transcript;
import java.util.Map;

public interface ResultAssembler {
  ExtractionResult assemble(
    String requestId,
    ExecutionPlan plan,
    Map<String, ResolvedField> resolvedFields,
    java.util.List<String> globalIssues,
    Map<String, Double> stageTimings,
    Transcript transcript,
    boolean includeTranscript,
    double processingTimeMs
  );
}

File: core/bin/main/com/gonezo/audioextraction/application/pipeline/ResultAssembler.kt
package com.gonezo.audioextraction.application.pipeline

import com.gonezo.audioextraction.domain.contract.ExtractionResult
import com.gonezo.audioextraction.domain.model.ExecutionPlan
import com.gonezo.audioextraction.domain.model.ResolvedField
import com.gonezo.audioextraction.domain.model.Transcript

interface ResultAssembler {
  fun assemble(
    requestId: String,
    plan: ExecutionPlan,
    resolvedFields: Map<String, ResolvedField>,
    globalIssues: List<String>,
    stageTimings: Map<String, Double>,
    transcript: Transcript,
    includeTranscript: Boolean,
    processingTimeMs: Double,
  ): ExtractionResult
}
